pinot-tools/src/main/java/com/linkedin/pinot/tools/perf/QueryRunner.java

```diff
@@ -0,0 +1,586 @@
+  private int _numIntervalsToIncreaseQPS = 10;
+  @Option(name = "-brokerHost", required = false, metaVar = "<String>", usage = "Broker host name (default localhost).")
+  private String _brokerHost = "localhost";
+  @Option(name = "-brokerPort", required = false, metaVar = "<int>", usage = "Broker port number (default 8099).")
+  private int _brokerPort = 8099;
+  @Option(name = "-help", required = false, help = true, aliases = {"-h", "--h", "--help"},
+      usage = "Print this message.")
+  private boolean _help;
+
+  @Override
+  public boolean getHelp() {
+    return _help;
+  }
+
+  @Override
+  public String getName() {
+    return getClass().getSimpleName();
+  }
+
+  @Override
+  public String description() {
+    return "Run queries from a query file in singleThread, multiThreads, targetQPS or increasingQPS mode.";
+  }
+
+  @Override
+  public boolean execute()
+      throws Exception {
+    if (!new File(_queryFile).isFile()) {
+      LOGGER.error("Argument queryFile: {} is not a valid file.", _queryFile);
+      printUsage();
+      return false;
+    }
+    if (_numIntervalsToReportClientTimeStatistics <= 0) {
+      LOGGER.error("Argument numIntervalsToReportClientTimeStatistics should be a positive number.");
+      printUsage();
+      return false;
+    }
+
+    LOGGER.info("Start query runner targeting broker: {}:{}", _brokerHost, _brokerPort);
+    PerfBenchmarkDriverConf conf = new PerfBenchmarkDriverConf();
+    conf.setBrokerHost(_brokerHost);
+    conf.setBrokerPort(_brokerPort);
+    conf.setRunQueries(true);
+    conf.setStartZookeeper(false);
+    conf.setStartController(false);
+    conf.setStartBroker(false);
+    conf.setStartServer(false);
+
+    switch (_mode) {
+      case "singleThread":
+        LOGGER.info("MODE singleThread with queryFile: {}, numIntervalsToReportClientTimeStatistics: {}", _queryFile,
+            _numIntervalsToReportClientTimeStatistics);
+        singleThreadedQueryRunner(conf, _queryFile, _numIntervalsToReportClientTimeStatistics);
+        break;
+      case "multiThreads":
+        if (_numThreads <= 0) {
+          LOGGER.error("For multiThreads mode, argument numThreads should be a positive number.");
+          printUsage();
+          break;
+        }
+        if (_reportIntervalMs <= 0) {
+          LOGGER.error("For multiThreads mode, argument reportIntervalMs should be a positive number.");
+          printUsage();
+          break;
+        }
+        LOGGER.info("MODE multiThreads with queryFile: {}, numThreads: {}, reportIntervalMs: {}, "
+                + "numIntervalsToReportClientTimeStatistics: {}", _queryFile, _numThreads, _reportIntervalMs,
+            _numIntervalsToReportClientTimeStatistics);
+        multiThreadedQueryRunner(conf, _queryFile, _numThreads, _reportIntervalMs,
+            _numIntervalsToReportClientTimeStatistics);
+        break;
+      case "targetQPS":
+        if (_numThreads <= 0) {
```

Comment:
Does args4j have check for valid values. If so, we can use that here, as well as other similar places.